effekt/shared/src/main/scala/effekt/lifted/Tree.scala

```diff
@@ -120,72 +118,70 @@ def Here() = Evidence(Nil)
 
 class EvidenceSymbol() extends Symbol { val name = Name.local(s"ev${id}") }
 
-def freeVariables(d: Definition): Set[Symbol] = d match {
+def freeVariables(d: Definition): Set[Param] = d match {
```

Comment:
Not sure collecting params here is the right thing to do. Eventually, we might drop param and replace it by separate lists of value and block bindings (like in core). Is it because you want to collect `(Set[ValueSymbol], Set[BlockSymbol])`?

Reply:
On the other hand, also having the types does make sense.

Reply:
Yes, having the types is the main reason, and that there is no type for evidence. An alternative would be a `List[(Id, Type)]` (or similar), but that would require us to add a type for Evidence (which would then ideally be used in other places, too.